messaging/src/main/java/org/axonframework/eventhandling/TrackingEventProcessor.java

```diff
@@ -175,7 +238,10 @@ protected void processingLoop(Segment segment) {
         BlockingStream<TrackedEventMessage<?>> eventStream = null;
         long errorWaitTime = 1;
         try {
-            while (state.get().isRunning() && canClaimSegment(segment.getSegmentId())) {
+            // only execute the loop when in running state, no processing instructions have been executed, and the
```

Comment:
Nit: I'd adjust this comment to:
`...state, no processing instructions have been executed **for the given Segment**, and the...`